src/components/create.vue

```diff
@@ -243,55 +243,54 @@ export default {
         return;
       }
+
+      let blockchain = getBlockchain(this.selectedChain);
+
       try {
-        apkey = PrivateKey.fromWif(this.activepk)
-          .toPublicKey()
-          .toString(this.selectedChain);
-        mpkey = PrivateKey.fromWif(this.memopk)
-          .toPublicKey()
-          .toString(this.selectedChain);
+        apkey = blockchain.getPublicKey(this.activepk);
+        mpkey = blockchain.getPublicKey(this.memopk);
         if (this.includeOwner == 1) {
-          opkey = PrivateKey.fromWif(this.ownerpk)
-            .toPublicKey()
-            .toString(this.selectedChain);
+          opkey =blockchain.getPublicKey(this.ownerpk);
         }
       } catch (e) {
+        console.error(e);
         this.errorMsg = this.$t("invalid_key_error");
         this.$refs.errorModal.show();
         return;
       }
       this.$refs.loaderAnimModal.show();
-      let verified = await Apis.instance(
-        this.$store.state.SettingsStore.settings.selected_node,
-        true
-      ).init_promise.then(() => {
-        return Apis.instance()
-          .db_api()
-          .exec("get_full_accounts", [[this.accountname], false])
-          .then(res => {
-            // TODO: Better verification
-            if (
-              res[0][1].account.active.key_auths[0][0] == apkey &&
-              (res[0][1].account.owner.key_auths[0][0] == opkey ||
-                this.includeOwner == 0) &&
-              res[0][1].account.options.memo_key == mpkey
-            ) {
-              this.$refs.loaderAnimModal.hide();
-              return res[0][1].account.id;
-            } else {
-              this.$refs.loaderAnimModal.hide();
-              this.$refs.errorModal.show();
-              this.errorMsg = this.$t("unverified_account_error");
-              return null;
-            }
-          });
+
+      blockchain.getAccount(this.accountname).then((account) => {
+          console.log(account);
+        let active_check = false;
+        account.active.public_keys.forEach((key) => {
+          if (key[0] == apkey) {
+            active_check = true;
+          }
+        });
+        let owner_check = !this.includeOwner;
+        account.owner.public_keys.forEach((key) => {
+          if (key[0] == opkey) {
+              owner_check = true;
+          }
+        });
+        let memo_check = account.memo.public_key == mpkey;
+        if (active_check && owner_check && memo_check) {
+          this.$refs.loaderAnimModal.hide();
+          this.accountID = account.id;
+          this.step = 3;
+        } else {
+          this.$refs.loaderAnimModal.hide();
+          this.$refs.errorModal.show();
+          this.errorMsg = this.$t("unverified_account_error");
+          this.accountID = "";
+        }
+      }).catch((err) => {
+          this.$refs.loaderAnimModal.hide();
+          this.$refs.errorModal.show();
+          this.errorMsg = this.$t("unverified_account_error");
+          this.accountID = "";
```

Comment:
See comment about making keys structure part of the chain config. Probably move the check to some kind of verifyKeys(...keys) method inside the blockchain implementation instead.

Reply:
Agreed, right now let's leave with you use the chain you login with, other than for verifying.